opennms-services/src/main/java/org/opennms/netmgt/collectd/SnmpCollectionSetRefactorDummy.java

```diff
@@ -0,0 +1,105 @@
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.collectd;
+
+import org.opennms.netmgt.collection.api.AttributeGroup;
+import org.opennms.netmgt.collection.api.CollectionAttribute;
+import org.opennms.netmgt.collection.api.CollectionResource;
+import org.opennms.netmgt.collection.api.CollectionSet;
+import org.opennms.netmgt.collection.support.builder.CollectionSetBuilder;
+import org.opennms.netmgt.collection.support.builder.GenericTypeResource;
+import org.opennms.netmgt.collection.support.builder.InterfaceLevelResource;
+import org.opennms.netmgt.collection.support.builder.NodeLevelResource;
+import org.opennms.netmgt.collection.support.builder.Resource;
+import org.opennms.netmgt.snmp.Collectable;
+import org.opennms.netmgt.snmp.proxy.LocationAwareSnmpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Collection;
+
+/**
+ * Extension to SnmpCollectionSet to create a CollectionSet with the CollectionsetBuilder. This is an immediate step
+ * in order to refactor the SnmpCollertor to use the CollectionsetBuilder.
+ */
+public class SnmpCollectionSetRefactorDummy extends SnmpCollectionSet implements Collectable, CollectionSet {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SnmpCollectionSetRefactorDummy.class);
+
+    SnmpCollectionSetRefactorDummy(SnmpCollectionAgent agent, OnmsSnmpCollection snmpCollection, LocationAwareSnmpClient client) {
+        super(agent,  snmpCollection, client);
+    }
+
+    CollectionSet createCollectionSet() {
+        CollectionSetBuilder builder = new CollectionSetBuilder(getAgent());
+
+
+        // TODO Patrick Very ugly mapping until refactoring is done. We convert here the snmp world into the CollectionSetBuilder world
+        for (CollectionResource collectionResource : getResources()) {
+            if(collectionResource instanceof NodeInfo) {
+                NodeInfo nodeInfo = (NodeInfo) collectionResource;
+                NodeLevelResource resource = new NodeLevelResource(nodeInfo.getNodeId());
+                addGroupsToBuilder(builder, resource, nodeInfo.getGroups());
+            }
+
+            if(collectionResource instanceof IfInfo) {
+                IfInfo ifInfo = (IfInfo) collectionResource;
+                InterfaceLevelResource resource = new InterfaceLevelResource(new NodeLevelResource(ifInfo.getNodeId()), ifInfo.getAttributesMap().get("snmpifname"));
+                addGroupsToBuilder(builder, resource, ifInfo.getGroups());
+            }
+
+            if(collectionResource instanceof AliasedResource) {
+                AliasedResource aliasedResource = (AliasedResource) collectionResource;
+                Resource resource = null; // TODO: Patrick there is no AliasedResource in the CollectionSetBuilder world, what shall we do?
```

Comment:
If the same behavior can't be achieved with one of the existing types, we'll need to add a new type.